src/goto-instrument/reachability_slicer.cpp

```diff
@@ -40,7 +40,8 @@ reachability_slicert::get_sources(
     if(criterion(e_it.first) || is_threaded(e_it.first))
       sources.push_back(e_it.second);
   }
-
+  if(sources.empty())
+    throw "no slicing criterion found";
```

Comment:
This might be a candidate for throwing https://github.com/diffblue/cbmc/blob/develop/src/util/exception_utils.h#L25
(which will be caught and reported by parse_options).

Reply:
Done.